pkg/wireguard/agent/agent.go

```diff
@@ -302,23 +330,49 @@ func (a *Agent) RestoreFinished(cm *clustermesh.ClusterMesh) error {
 		// Wait until we received the initial list of nodes from all remote clusters,
 		// otherwise we might remove valid peers and disrupt existing connections.
 		cm.NodesSynced(context.Background())
+		// Additionally wait for ipcache synchronization, so that we don't garbage
+		// collect non-stale AllowedIPs too early.
+		cm.IPIdentitiesSynced(context.Background())
 	}
 
 	a.Lock()
 	defer a.Unlock()
 
 	// Delete obsolete peers
 	for _, p := range a.peerByNodeName {
-		delete(a.restoredPubKeys, p.pubKey)
+		staleIPs := a.computeRestoredIPs(p.pubKey)
```

Comment:
Would it be more accurate to call these "duplicate" allowed IPs rather than "stale" allowed IPs?

Reply:
Hmm, IMO stale is more accurate here, as it refers to IPs of pods that disappeared while the agent was down. There's an duplicated IPs angle (to avoid that we assign the same IP to two peers upon restore), but that is orthogonal.